PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/ui/CosmosServerlessSparkSubmissionPanel.kt

```diff
@@ -82,19 +85,22 @@ open class CosmosServerlessSparkSubmissionPanel(private val project: Project)
     inner class ViewModel: SparkSubmissionContentPanel.ViewModel() {
         init {
             clusterSelection.clusterIsSelected
-                    .subscribe { cluster ->
-                        if (cluster == null) {
-                            return@subscribe
-                        }
-
-                        val model = CosmosServerlessSparkSubmitModel().apply {
-                            getData(this)
-                        }
-
-                        setData(model.apply {
-                            sparkEventsDirectoryPrefix = "adl://${cluster.name}.azuredatalakestore.net/"
-                        })
+                .flatMap { cluster ->
+                    if (cluster == null) {
+                        Observable.empty<CosmosServerlessSparkSubmitModel>()
+                    } else {
+                        Observable.just(CosmosServerlessSparkSubmitModel())
+                            .doOnNext { getData(it) }
+                            .map { model ->
+                                model.apply {
+                                    sparkEventsDirectoryPrefix = (cluster as? AzureSparkServerlessAccount)?.storageRootPath
```

Comment:
Is it the only line needed to change?

Reply:
upload path should also be changed and I put the change in `validateStorageInfo` method of class `SparkSubmissionJobUploadStorageWithUploadPathPanel`

---
In reply to: [244076765](https://github.com/Microsoft/azure-tools-for-java/pull/2530#discussion_r244076765) [](ancestors = 244076765)